spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationContextUtils.java

```diff
@@ -105,6 +106,8 @@
 
 	public static final String DISPOSABLES_BEAN_NAME = "integrationDisposableAutoCreatedBeans";
 
+	public static final String MESSAGE_HANDLER_FACTORY_BEAN_NAME = "integrationMessageHandlerMethodFactory";
```

Comment:
Wait. And where is the bean for this name?
What is the point then to deprecate that `HandlerMethodArgumentResolversHolder` if we don't provide a valuable alternative out-of-the-box?

Reply:
Well, we do by allowing injection of `integrationMessageHandlerMethodFactory` (see `HandlerMethodArgumentResolversHolder` deprecation comment)

Reply:
What do you mean _good argument about integrationMessageHandlerMethodFactory bean definition._?

Reply:
My point is that you just don't register the bean for the `integrationMessageHandlerMethodFactory` as out-of-the-box, default solution.
This way it doesn't sound reasonable to deprecate another feature since we really don't replace it.